datalad/tests/fixtures.py

```diff
@@ -0,0 +1,73 @@
+"""utilities to assist with setting up tests"""
+
+from datalad.api import clone
+from datalad.core.distributed.clone import decode_source_spec
+from datalad.distribution.dataset import Dataset
+from datalad.interface.common_cfg import dirs
+from datalad.utils import Path, better_wraps, optional_args
+from datalad.tests.utils import with_tempfile
+
+
+DATALAD_TESTS_CACHE = Path(dirs.user_cache_dir) / "tests"
+
+
+def get_cached_dataset(url, dataset_name=None, version=None):
+    """ Helper to get a cached clone from url
+
+    Intended for use from within `datalad_dataset` decorator.
+    Clones `url` into user's cache under datalad/tests/`name`. If such a clone
+    already exists, don't clone but return the existing one. So, it's supposed
+    to cache the original source in order to reduce time and traffic for tests,
+    by letting subsequent requests clone from a local location directly.
+
+    If it's an annex get the content, too.
```

Comment:
I would make it an option. May be make it a list of paths to get?
 Some tests might need just some files.
Add a note that dataset might or not already contain data for annexed files

Reply:
Yes. Will go for a list of paths, I guess.